utils/push_to_postgres.py

```
@@ -0,0 +1,259 @@
+            metric.get("command"),
+            metric.get("data_size"),
+            metric.get("pipeline"),
+            metric.get("clients"),
+            metric.get("requests"),
+            metric.get("rps"),
+            metric.get("avg_latency_ms"),
+            metric.get("min_latency_ms"),
+            metric.get("p50_latency_ms"),
+            metric.get("p95_latency_ms"),
+            metric.get("p99_latency_ms"),
+            metric.get("max_latency_ms"),
+            metric.get("cluster_mode"),
+            metric.get("tls"),
+            metric.get("io_threads"),
+            metric.get("valkey-benchmark-threads"),
+            metric.get("benchmark_mode"),
+            metric.get("duration"),
+            metric.get("warmup"),
+            metric.get("architecture"),
+        )
+        rows.append(row)
+    return rows
+
+
+def push_to_postgres(metrics_data, conn, dry_run=False):
+    """Push metrics to PostgreSQL."""
+    print(f"  Converting {len(metrics_data)} metrics to rows...")
+    rows = convert_metrics_to_rows(metrics_data)
+
+    if dry_run:
+        print(f"Would insert {len(rows)} rows:")
+        for i, row in enumerate(rows[:3]):
+            print(f"  [{i+1}] {row}")
+        if len(rows) > 3:
+            print(f"  ... and {len(rows) - 3} more")
+        return len(rows)
+
+    print(f"  Inserting {len(rows)} rows into database...")
+    with conn.cursor() as cur:
+        execute_values(
+            cur,
+            """
+            INSERT INTO benchmark_metrics (
+                timestamp, commit, command, data_size, pipeline, clients, requests,
+                rps, avg_latency_ms, min_latency_ms, p50_latency_ms, p95_latency_ms,
+                p99_latency_ms, max_latency_ms, cluster_mode, tls, io_threads,
+                benchmark_threads, benchmark_mode, duration, warmup, architecture
+            ) VALUES %s
+            ON CONFLICT (timestamp, commit, command, data_size, pipeline) DO NOTHING
```

Probably silently discarding rows here. Let's at least detect this and log it. (Or comment clearly if this is intentional, I suppose)